src/pages/test-report/components/competency/competency.ts

```diff
@@ -125,12 +117,58 @@ export class CompetencyComponent {
     }
   }
 
+  onTap = () => {
+    this.addOrRemoveFault();
+  }
+
+  onPress = () => {
+    this.addOrRemoveFault(true);
+  }
+
+  canButtonRipple = (): void => {
+    if (this.isRemoveFaultMode) {
+      if (this.hasDangerousFault && this.isDangerousMode) {
+        this.allowRipple = true;
+        return;
+      }
+
+      if (this.hasSeriousFault && this.isSeriousMode) {
+        this.allowRipple = true;
+        return;
+      }
+
+      if (!this.isSeriousMode && !this.isDangerousMode && this.faultCount > 0) {
+        this.allowRipple = true;
+        return;
+      }
+      this.allowRipple = false;
+    } else {
+      if (this.hasDangerousFault) {
+        this.allowRipple = false;
+        return;
+      }
+
+      if (this.isDangerousMode) {
+        this.allowRipple = true;
+        return;
+      }
+
+      if (this.hasSeriousFault) {
+        this.allowRipple = false;
+        return;
+      }
+
+      if (this.isSeriousMode) {
+        this.allowRipple = true;
+        return;
+      }
+      this.allowRipple = true;
+    }
+  }
+
```

Comment:
I kept this purposely verbose to make it easier to read, but open to ideas on how it could be improved.

Reply:
Have we got some unit tests for this logic?

Reply:
I think this function could be extracted somewhere where we just give a few parameters and it gives us back a boolean value. And it could be used for enabling or disabling ripple on other buttons as well. But not sure how, so let's just go with this for now and have a think about it in the future.

Reply:
Not yet... I'm on it

Reply:
> I think this function could be extracted somewhere where we just give a few parameters and it gives us back a boolean value. And it could be used for enabling or disabling ripple on other buttons as well. But not sure how, so let's just go with this for now and have a think about it in the future.

Agreed

Reply:
> I think this function could be extracted somewhere where we just give a few parameters and it gives us back a boolean value. And it could be used for enabling or disabling ripple on other buttons as well. But not sure how, so let's just go with this for now and have a think about it in the future.

I feel the same. It's replicating some of the logic in addFault / removeFault and feels like it could be done better.

Reply:
Added tests. Will do similar for Controlled Stop in next PR.